solution/index.js

```diff
@@ -0,0 +1,704 @@
+      
+        }).catch(error =>{
+            alert("There seems to be a problem");
+            console.log(error);
+        }); 
+        if(response.status === 200 || response.status === 418){
+            alert("You Saved Successfully To The API");
+        }
+}
+
+
+function deleteConfirmation(){
+    if(confirm("Proceeding will delete tasks from the screen and the local storage, Are You Sure ? "))
+    clearAllData();
+
+    else alert("That Was Close ...")
+       
+    
+}
+
+/* clears from DOM and local storage */
+ function clearAllData(){
+   
+    clearTodoList();
+    clearProgressList();
+    clearDoneList();
+    
+}
+
+
+function clearTodoList(){
+    
+    tasksStorage.todo = [];
+    updateStorage();
+         /* removing from DOM */
+    while(todolistElement.firstChild) todolistElement.removeChild(todolistElement.firstChild);
+}
+
+function clearProgressList(){
+   
+    tasksStorage["in-progress"] = [];
+    updateStorage();
+            /* removing from DOM */
+    while(progressListElement.firstChild) progressListElement.removeChild(progressListElement.firstChild);
+}
+
+function clearDoneList(){
+   
+    tasksStorage.done = [];
+    updateStorage();
+                /* removing from DOM */
+    while(doneListElement.firstChild) doneListElement.removeChild(doneListElement.firstChild);
+}
+
+
+/* ---- event listeners ---- */
+container.addEventListener("click", handleAddTaskEvent); // add task with event delegation
+    
+document.addEventListener('focus', getOriginalText, true); // for changing text
+
+document.addEventListener('blur', handleTaskContentEdit, true); // for changing text
+
+document.addEventListener("keydown", listShortcuts)
+
+document.addEventListener("keyup", settingPressedKeys);  // for moving and deleting tasks
+
+
+saveButton.addEventListener("click", saveToAPI);    // save button
+
+loadButton.addEventListener("click", loadFromAPI);  // load button
+
+clearButton.addEventListener("click", deleteConfirmation);    // clear all button
+
+    /* clear List Buttons */
+clearToDoListBtn.addEventListener("click", clearTodoList)
+
+clearProgressListBtn.addEventListener("click",clearProgressList)
+
+clearDoneListBtn.addEventListener("click", clearDoneList)  
+
+
+ /* The function to start the show */
+ initialFromLocalStorage();
+ 
+
+
+ /* -- drag and drop -- */
+ const dropZones = document.querySelectorAll("ul"); // use for drag and drop
+
+
+ document.addEventListener("dragstart", (e) => { 
+    e.dataTransfer.setData("text/plain", e.target.id);   // dragged element id transferd
+     currentList =  parentList(e.target.parentElement.id);
+})
+
+// when task is over a list
+for(const drop of dropZones) {
+    drop.addEventListener("dragover", (e) => {
+        e.preventDefault();
+        
+    })
+
+
+    // when task is dropped
+    drop.addEventListener("drop", (e) => {
```

Comment:
You get into a complication when you insert the event into the for loop. The functions should stand separately.